lib/currentProfile.js

```diff
@@ -0,0 +1,136 @@
+'use strict';
+
+
+// API for client
+// currently no support to the past
+var record = null;
+var updateHandler = null;
+
+	function updateFromStorage(profileStorage) { 
+		profileStorage.last( function(err, records) {
+			record = records.length > 0 ? records[0] : null;
+			if (record) {
+				console.log('Profile record loaded from storage.');
+				if (updateHandler) updateHandler(record);
+			}
+		});
+	}
+	
+	function update(newrecord) { 
+		record = newrecord;
+		if (updateHandler) updateHandler(record);
+	}
+	
+	function dia() { 
+		return record ? record.dia : 3;
+	}
+	
+	function ic (time) {
+		if (!record) return null;
+		if (!record.calculator) return record.carbratio;
+		var minutes =  time.getMinutes() + time.getHours() * 60;
+		for (var i=0; i < record.ic.length -1; i++) {
+			if (parseInt(record.ic[i+1].from)>minutes) return record.ic[i].val;
+		}
+		return null;
+	}
+
+	function isf (time,units) {
+		if (!record) return 0;
+		if (!record.calculator) return convertUnits(record.sens,units);
+		var minutes =  time.getMinutes() + time.getHours() * 60;
+		for (var i=0; i < record.isf.length -1; i++) {
+			if (parseInt(record.isf[i+1].from)>minutes) return convertUnits(record.isf[i].val,units);
+		}
+		return null;
+	}
+
+	function car(time, glycemicIndex) { //glycemicIndex 1..low  2..medium 3..high but allowing float within range <1-3>
+		if (!record) return null;
+		if (record.simple) return record.carbs_hr;
+		if (glycemicIndex==1) return record.car_low;
+		if (glycemicIndex==2) return record.car_medium;
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Expected '{' and instead saw 'return'.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769498805/issues/source?bid=2452374&fileBranchId=2459081#l52)